modules/photo/src/seamless_cloning.cpp

```diff
@@ -68,17 +68,30 @@ void cv::seamlessClone(InputArray _src, InputArray _dst, InputArray _mask, Point
 {
     CV_INSTRUMENT_REGION();
     CV_Assert(!_src.empty());
+    CV_Assert(!_dst.empty());
 
     const Mat src  = _src.getMat();
     const Mat dest = _dst.getMat();
+
+    if (p == Point())
+    {
+        p.x = dest.size().width / 2;
+        p.y = dest.size().height / 2;
+    }
```

Comment:
The `p` parameter does not have default value like `Point p = Point()`. It means that the point is user provided value. I do not think that we sound handle `Point()`, a.k.a. (0,0) as special case.

Reply:
what about using Point(-1,-1) to centralize as special case ?

Reply:
Why do we need special cases in code? User should provide location, shouldn't he/she?

Reply:
"In issue #15928, the user reports:

>>I had:

```
src.shape   = 500,500
src_mask    = 500,500
dest.shape  = 500,500

h, w = src_mask.shape[:2]
```
>>At this point, I didn't see the need for the required argument `p`, as I expected the function to automatically match the mask. However, since `p` is a required argument, I provided `(w//2, h//2)` for `p` (the center of the mask). But instead of aligning the source image (`src`) into the destination (`dest`), the resulting cloning was shifted up to the center of the screen!
It seems like there may have been an issue with how the center point was calculated or used. The user expected the function to align the source (src) based on the mask (src_mask) automatically, but instead, the function seemed to shift the cloning to an unexpected location.

Perhaps the logic for finding the center point could be improved or made more intuitive to help avoid confusion in such cases. A clearer method of determining the center, or perhaps some additional documentation on how to properly use the p argument, might help facilitate this process."

Reply:
also in the issue the user commented 
>>I got the expected behaviour by using a rect of thickness 2

```
h, w = src_mask.shape[:2]
# rect of thickness > 1
cv2.rectangle(src_mask,(0,0), (w, h) ,255, 2)
# the clone will now aligns from the centre
out=cv2.seamlessClone(src, dest, src_mask,(w//2, h//2),flags= cv2.NORMAL_CLONE)
```
a rect of 1 thickness did not work.

Reply:
i will revert this idea if you want.

Reply:
there is related issues on some repos like https://github.com/OpenTalker/SadTalker/issues/542
let me investigate if

```
    {
        p.x = dest.size().width / 2;
        p.y = dest.size().height / 2;
    }
```
works perfectly fine for different dimensions